### PR TITLE
Source Code Disclosure CVE20121823 only analyze JS on LOW threshold

### PR DESCRIPTION
**Pull request by psiinon**
With added tests :D
 
Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Updated to fix the conflict

File: addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureCVE20121823.java [hunk 1]
```diff
@@ -134,6 +134,10 @@ public void scan() {
             if (!getBaseMsg().getResponseHeader().isText()) {
                 return; // Ignore images, pdfs, etc.
             }
+            if (getAlertThreshold() != AlertThreshold.LOW
```
[the remark this answers is not in the document]
Comment (psiinon) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureCVE20121823.java]: I think both should work: https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals
Not sure which one we should standardize on...

[the remark this answers is not in the document]
Comment (psiinon) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureCVE20121823.java]: Looks like == might be slightly better as its null safe ;)

[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureCVE20121823.java]: Ok

**Pull request merged**